### PR TITLE
Feat: REPLAT-2167 Topic name query

### PR DESCRIPTION
**Pull request by jimhunty**
This takes the work @jackharvey1 has done on TPA to expose Name and Description in a separate request and wraps the Topic component in the same way that Author Profile does. This means our Name and Description are data-driven.

### CODE REVIEW

**Comment by times-tools** [comment]
If you use Expo, view our components by scanning this qr code: <br>  <br> This has been made possible through [Fructose](https://github.com/newsuk/fructose)

**Comment by L0wry** [comment]
rebasing should fix dextrose

File: packages/topic/__tests__/topic-functional.js [hunk 1]
```diff
@@ -4,10 +4,12 @@ import { fixtureGenerator } from "@times-components/provider-test-tools";
 import { delay, MockedProvider } from "@times-components/utils";
 import Topic from "../src/topic";
 
-jest.mock("@times-components/article-list", () => "ArticleList");
+// eslint-disable-next-line global-require
+jest.mock("@times-components/article-list", () => require("./articleListMock"));
```
Comment (jackharvey1) [inline packages/topic/__tests__/topic-functional.js]: does `jest.mock('./blah', () => ({ default: "ArticleList", ArticleListPageError: "ArticleListPageError" }));` not work?

**Comment by jackharvey1** [comment]
Should we be updating the provider rather than adding another? Domain knowledge on this is still fuzzy but with this pattern aren't we going to require two providers feeding into the `TopicPage` component? Additionally this is two API calls when it could be one.

**Comment by times-tools** [comment]
If you use Expo, view our components by scanning this qr code: <br>  <br> This has been made possible through [Fructose](https://github.com/newsuk/fructose)

**Comment by jimhunty** [comment]
@jackharvey1 Possibly. At the moment, I have just been copying the example set out in Author profile. If this is the case then we can look to refactor later.

**Comment by jackharvey1** [comment]
@jimhunty should we not just do it now?

File: packages/provider/src/provider.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ import AuthorArticlesWithImagesProvider, {
   query as articleListWithImagesQuery
 } from "./author-articles-with-images";
 import ArticleProvider, { query as articleQuery } from "./article";
+import TopicProvider, { query as topicQuery } from "./topic";
```
Comment (jackharvey1) [inline packages/provider/src/provider.js]: should we call this `topic-head`? In my head topic-head and topic-articles are siblings whose parent is both topic.

File: packages/topic/__tests__/topic-functional.js (hunk 1 above)
Comment (colincclark) [inline packages/topic/__tests__/topic-functional.js]: 👍

Comment (jackharvey1) [inline packages/topic/__tests__/topic-functional.js]:
```
jest.mock("@times-components/article-list", () => ({
  default: "ArticleList",
  ArticleListPageError: "ArticleListPageError"
}));
```
works for me

File: packages/topic/__tests__/topic-functional.js [hunk 3]
```diff
@@ -55,12 +57,18 @@ export default () => {
   });
 
   it("should render the loading state", () => {
+    const tree = renderer.create(<Topic {...props} isLoading />);
+
+    expect(tree).toMatchSnapshot("2. Render an topics page loading state");
```
Comment (colincclark) [inline packages/topic/__tests__/topic-functional.js]: render _a_ topics page

File: packages/provider/src/provider.js (hunk 2 above)
Comment (jimhunty) [reply packages/provider/src/provider.js]: Again this was following the naming conventions set in Author Profile. If we change this here, we should change it there as well. Same with the provider as mentioned above. Something we should tackle in a separate ticket

File: packages/topic/__tests__/topic-functional.js (hunk 1 above)
Comment (jimhunty) [reply packages/topic/__tests__/topic-functional.js]: Still doesn't work for me locally

**Comment by times-tools** [comment]
If you use Expo, view our components by scanning this qr code: <br>  <br> This has been made possible through [Fructose](https://github.com/newsuk/fructose)

**Pull request merged**